x/gamm/types/route.go

```diff
@@ -21,6 +28,10 @@ func (routes SwapAmountInRoutes) Validate() error {
 
 type SwapAmountOutRoutes []SwapAmountOutRoute
 
+func (routes SwapAmountOutRoutes) IsOsmoRoutedMultihop() bool {
+	return len(routes) == 2 && (routes[1].TokenInDenom == appparams.BaseCoinUnit)
```

Comment:
Should this support routes longer than 2? We could check that every odd route passes `r.TokenInDenom == appparams.BaseCoinUnit`

Reply:
n general the idea that osmo will have a lot of pools X-OSMO
Then to Swap X->Y all swaps will looks like X -> OSMO -> Y
So to detect such swaps we take the first hop and check that it has output OSMO(that mean that input for the second hop will be osmo too)

    What if route has length of three which all includes Osmo in the routes

That possible only if we have second hop OSMO->OSMO, or i miss something?

Reply:
I was thinking of X -> OSMO -> Y -> OSMO -> Z.

Those can probably be simplified to less hops if there are x/osmo and z/osmo pools, but it isn't guaranteed that they exist.

I don't have a very strong opinion here, just thought it would make sense to future proof the iteration

Reply:
`X -> OSMO -> Y -> OSMO -> Z` would imply that there are two `Y/OSMO` pools?  Which I think is a case we don't really have to worry about rn

Reply:
Also, knowing that there is such mechanic in place, tokens will be listed to OSMO and immediately will be available to all other assets for swaps with discount.